riscv/src/compiler.rs

```diff
@@ -566,6 +566,7 @@ fn preamble(degree: u64, coprocessors: &CoProcessors) -> String {
     instr jump l: label { pc' = l }
     instr load_label l: label -> X { X = l }
     instr jump_dyn X { pc' = X }
+    instr jump_dyn_if_nonzero X { pc' = X * XIsZero + pc + 1 * (1 - XIsZero) }
```

Comment:
Is this really correct?
If X is zero, this equals 0 * 1 + pc + 1 * (1 - 1) = 0.
If X = 1, 1 * 0 + pc + 1 * (1 - 0) = pc + 1. Shouldn't this jump to X?

Reply:
Yeah, I'm not sure what happened there :/ Should be fixed now, now the constraint is very similar to `branch_if_nonzero`.